Submitted By:     VMiklos <mamajom@axelero.hu>
Date:             2005.01.07
Initial Packages: 2.8.1
Description:      This patch adds a new feature to makepkg.
                  Before starting the build, checks for a newer version of the
                  given package available.

diff -Naur pacman-2.8.1.orig/scripts/makepkg pacman-2.8.1/scripts/makepkg
--- pacman-2.8.1.orig/scripts/makepkg	2004-07-17 03:55:28.000000000 +0200
+++ pacman-2.8.1/scripts/makepkg	2004-07-19 10:37:11.000000000 +0200
@@ -377,6 +377,28 @@
 
 cd $startdir
 
+# version checking
+if [ "x$up2date" == "x" ]; then
+	warning "Version checking informations are missing. Cannot check for newer version."
+else
+	msg "Checking for newer version..."
+	errors=0
+	if [ "$pkgver" != "$up2date" ]; then
+		echo "   FAILED"
+		if [ `type -p lynx` ]; then
+			errors=1
+		else
+	    		warning "The lynx program is missing.  Cannot check for newer version!"
+		fi
+	elif [ "$pkgver" == "$up2date" ]; then
+	        echo "    Passed"
+	fi
+	if [ $errors -gt 0 ]; then
+		error "Newer version available! ($pkgver < $up2date)"
+		exit 1
+	fi
+fi
+
 # retrieve sources
 msg "Retrieving Sources..."
 mkdir -p src
